Updated docker dependencies.

mgoltzsche opened the pull request:
Previous docker/docker/* dependency versions replaced with the corresponding versions from the latest checked docker's vendor.conf.
Added containerd/continuity dep since it is required by docker code.

Signed-off-by: Max Goltzsche <max.goltzsche@gmail.com>

mgoltzsche commented:
Hm, the skopeo test fails: The same dependencies must also be updated in skopeo because the `dockerclient.ParseHost()` call changed to `dockerclient.ParseHostURL()` ... another evening ...

rhatdan commented:
@mgoltzsche Are you still working on this?

rhatdan commented:
@runcom @mtrmac PTAL

mgoltzsche commented:
Finally it should work like this:
1) Merge #446
2) Merge projectatomic/skopeo#498
3) Merge #452

mgoltzsche commented:
Ah, it cannot yet be merged since skopeo integration test fails at tls verification...

mgoltzsche commented:
Fixed now - ready to be merged

TomSweeneyRedHat commented:
@mgoltzsche looks like you need to rebase.

mgoltzsche commented:
@TomSweeneyRedHat done

mtrmac commented:
Thanks, and I apologize for the late response.

I was going to carry this PR and merge it to avoid another iteration, but I have noticed that several of the vendored dependencies are switched to older versions; why? At least some of them are switched to exactly the versions vendored by `docker/docker`, which could be fair enough (but seems unnecessary, when even the older `docker/docker` version we had was working with the newer versions fine), but e.g. `docker-credential-helpers` is not used in any of the other packages AFAICT but still was reverted one commit back. What’s the goal?

The process we usually use for updating both containers/image and skopeo is a bit simpler, see https://github.com/projectatomic/skopeo#dependencies-management . Two PRs are enough, the temporary CI breakage in this repo is accepted. So, I’ll close the third PR.

mgoltzsche commented:
Sorry, previously I did it the other way around. Now I updated this PR so that it can be merged immediately since everything but the skopeo test passes. The latter works when this PR is merged as shown in the successful projectatomic/skopeo#498 build which uses this PR's `docker_dependency_update` branch.

Also previously I simply copied the dependencies from docker's vendor.conf which is why I accidentally downgraded some. Now I only updated `github.com/docker/docker` and `github.com/docker/go-connections` and added `github.com/containerd/continuity`since it is required by docker/docker

When this PR is merged I'll let projectatomic/skopeo#498 refer to containers/image master again.

mgoltzsche commented:
Sorry, previously I did it the other way around. Now I updated this PR so that it can be merged immediately since everything but the skopeo test passes. The latter works when this PR is merged as shown in the successful projectatomic/skopeo#498 build which uses this PR's `docker_dependency_update` branch.

Also previously I simply copied the dependencies from docker's vendor.conf which is why I accidentally downgraded some. Now I only updated `github.com/docker/docker` and `github.com/docker/go-connections` and added `github.com/containerd/continuity`since it is required by docker/docker

When this PR is merged I'll let projectatomic/skopeo#498 refer to containers/image master again.

mtrmac commented:
👍 

Thanks! (I’ll carry https://github.com/projectatomic/skopeo/pull/498 and merge it now to avoid another round trip.)
